### ISSUE TITLE
Xcode 8 (beta 3) Analyzer warning: Null is returned from a method that is expected to return a non-null value

### ISSUE THREAD

**Issue by funnel20** [AFNetworking/AFNetworking#3619]
When running the Analyzer in Xcode 8 beta 3, I get this warning:
_Null is returned from a method that is expected to return a non-null value_

on `return uploadTask;` in `uploadTaskWithRequest:fromFile:progress:completionHandler:` of AFURLSessionManager.m

**Comment by fishandphil** [comment]
We'd love this to be fixed as well :)
<img width="1099" alt="screen shot 2017-02-14 at 15 21 40" src="https://cloud.githubusercontent.com/assets/1768773/22935137/833f2d02-f2c9-11e6-8524-89a7161700a6.png">

**Comment by funnel20** [comment]
This is still present in Xcode 9.1 and very annoying as we strive to ship an app without any Analyzer warnings.
Is there any suggestion to solve this?

**Comment by funnel20** [comment]
Based on another issue https://github.com/AFNetworking/AFNetworking/issues/1675 I found a solution in another branch: https://github.com/Ner1Co/AFNetworking/commit/d11ac9618573e90b72d8c414d57d7276fe06dd0f

This will change the code to:
```ObjC
- (NSURLSessionUploadTask *)uploadTaskWithRequest:(NSURLRequest *)request
                                         fromFile:(NSURL *)fileURL
                                         progress:(void (^)(NSProgress *uploadProgress)) uploadProgressBlock
                                completionHandler:(void (^)(NSURLResponse *response, id responseObject, NSError *error))completionHandler
{
    __block NSURLSessionUploadTask *uploadTask = nil;
    url_session_manager_create_task_safely(^{
        uploadTask = [self.session uploadTaskWithRequest:request fromFile:fileURL];

        // uploadTask may be nil on iOS7 because uploadTaskWithRequest:fromFile: may return nil despite being documented as nonnull (https://devforums.apple.com/message/926113#926113)
        if (!uploadTask && self.attemptsToRecreateUploadTasksForBackgroundSessions && self.session.configuration.identifier) {
            for (NSUInteger attempts = 0; !uploadTask && attempts < AFMaximumNumberOfAttemptsToRecreateBackgroundSessionUploadTask; attempts++) {
                uploadTask = [self.session uploadTaskWithRequest:request fromFile:fileURL];
            }
        }
    });

    [self addDelegateForUploadTask:uploadTask progress:uploadProgressBlock completionHandler:completionHandler];

    return uploadTask;
}
```

So the IF statement is relocated within `url_session_manager_create_task_safely()`.
This will remove the Xcode Analyzer warning.
Is this the solution?

**Comment by funnel20** [comment]
@SlaunchaMan Hi, it's great to finally see a new official release (3.2.0).
It would be great to have these Xcode warnings solved as well.
Can I contribute by creating a PR?

**Comment by funnel20** [comment]
See PR https://github.com/AFNetworking/AFNetworking/pull/4122

**Comment by funnel20** [comment]
I can confirm this issue has been resolved in version 3.2.1. Thanks!
